lib/cppcheck.cpp

```diff
@@ -659,8 +659,23 @@ unsigned int CppCheck::check(const std::string &path, const std::string &content
     return checkFile(Path::simplifyPath(path), emptyString, &iss);
 }
 
+void CppCheck::fileNotFoundError(const std::string& filename)
+{
+    mErrorLogger.reportErr(ErrorMessage({ErrorMessage::FileLocation(filename, 0, 0)},
+                                        filename,
+                                        Severity::error,
+                                        "File not found",
+                                        "fileNotFound",
+                                        Certainty::normal));
+    mExitCode = 1;
+}
+
 unsigned int CppCheck::check(const ImportProject::FileSettings &fs)
 {
+    if (!Path::isFile(fs.filename)) {
```

Comment:
Other projects perform these checks with `importproject.cpp` so this duplicates logic. Having an controllable logic might be better though. But we should implement and this properly this properly for all projects. We should not introduce more half-baked or inconsistent behavior. We still trying to get rid of such things from years and years ago.

I also think this might be the wrong location to do this. I think this should be reported closer to the actual import but I think we might be lacking common code to place this in.

Reply:
We can put it in importproject also. But I wanted to achieve that the analysis will continue on other files that are found. And this check here means that if the file has been removed after the project was loaded then we still detect this problem and bails out with an error message. Still.. maybe we can tune this and report it if simplecpp fails to read the source file instead.

Reply:
I understood the intention. I just want it to be consistent in implementation and test.

We should do in in the import step as we do with `Path::acceptFile()` so we know this as early as possible and not bail out on the first error. Similar to the changes I did for library and addon loading.

The problem is that we are not able to collect errors there - yet. That's what I added `CmdLineLogger` for. I want to collect all errors and report them with a proper ID after all options have been processed. But I need a few more things merged and changed first.

So this change is fine for now as we cannot solve it in a better way yet.

Reply:
I will post a draft PR soon after my pending stuff has been merged and I finished up most of the executor stuff. If things go well that should be as soon as next week.

Reply:
> I just want it to be consistent in implementation and test.

hmm yes

> I will post a draft PR soon after my pending stuff has been merged and I finished up most of the executor stuff.

ok sure. 

> If things go well that should be as soon as next week.

I would like to have this in next major release 2.13 so well it's not a hurry yet.

Reply:
> > I just want it to be consistent in implementation and test.
> 
> hmm yes

Simply adding tests with missing files for all projects would be sufficient.